docs/resources/swr_repository.md

```diff
@@ -0,0 +1,66 @@
+---
+subcategory: "Software Repository for Container (SWR)"
+---
+
+# huaweicloud_swr_repository
+
+Manages a SWR repository resource within HuaweiCloud.
+
+## Example Usage
+
+```hcl
+variable "organization_name" {} 
+
+resource "huaweicloud_swr_repository" "test" {
+  organization = var.organization_name
+  name         = "%s"
+  description  = "Test repository"
+  category     = "linux"
+  is_public    = false
+}
+```
+
+## Argument Reference
+
+The following arguments are supported:
+
+* `region` - (Optional, String, ForceNew) Specifies the region in which to create the resource. If omitted, the
+  provider-level region will be used. Changing this creates a new resource.
+
+* `organization` - (Required, String, ForceNew) Specifies the name of the organization (namespace) the repository belongs.
+  Changing this creates a new resource.
+
+* `name` - (Required, String, ForceNew) Specifies the name of the repository. Changing this creates a new resource.
+
+* `is_public` - (Required, Bool) Specifies whether the repository is public.
```

Comment:
suggest to make the param be optional and defaults to `false`

Reply:
fixed